NTPClient.h

```diff
@@ -5,9 +5,10 @@
 
 class NTPClient {
     public:
-        NTPClient(NetworkInterface *iface);
+        explicit NTPClient(NetworkInterface *interface = NULL);
```

Comment:
I have to admit, I haven't come across the use of the `explicit` keyword all that much. Could you clarify its purpose here?

Reply:
my [c++ linter](http://cppcheck.sourceforge.net/) suggested that and also yesterday i have read [this post](https://www.acodersjourney.com/top-25-cplusplus-api-design-mistakes-and-how-to-avoid-them/) `Mistake #6` so i thought i would use it

Reply:
Thanks for the links, I found this article to be helpful as well: http://www.cs.technion.ac.il/users/yechiel/c++-faq/explicit-ctors.html.